meta-balena-common/recipes-core/initrdscripts/files/resindataexpander

```diff
@@ -20,11 +22,26 @@ resindataexpander_enabled() {
 }
 
 resindataexpander_run() {
-    info "resindataexpander: Expand extended partition... "
-    parted -s /dev/$datadev -- resizepart 4 -1s
-    info "resindataexpander: Finished expanding extended partition."
+    case "$(blkid -o value -s PTTYPE /dev/$datadev)" in
+        dos)
+            extpart_num=$(echo "$parted_print"  | grep ":::lba" | tail -1 | cut -d: -f1)
+            datapart_num=$(expr $extpart_num \+ 2)
+            ;;
+        gpt)
+            datapart_num=$(echo "$parted_print"  | grep "resin-data" | tail -1 | cut -d: -f1)
+            ;;
+        *)
+            fatal "Unrecognized partition table type."
+            ;;
+    esac
+
+    if [ -n "$extpart_num" ]; then
+        info "resindataexpander: Expand extended partition... "
+        parted -s /dev/$datadev -- resizepart $extpart_num 100%
+        info "resindataexpander: Finished expanding extended partition."
+    fi
     info "resindataexpander: Expand data partition... "
-    parted -s /dev/$datadev -- resizepart 6 -1s
+    parted -s /dev/$datadev -- resizepart $datapart_num 100%
```

Comment:
resin-data-expander has now removed the hardcoding of partitions making it suitable for other partition table formats. See https://github.com/balena-os/meta-balena/blob/26f67abe90faf461ba9352bc5b7627e6a250165e/meta-balena-common/recipes-core/initrdscripts/files/resindataexpander